### ISSUE TITLE
Moved from NeDB to MongoDB on Windows issue

### ISSUE THREAD

**Issue by PathfinderNetworks** [Ylianst/MeshCentral#2062]
I just moved from NeDB to MongoDB on my Server 2019 box running 0.7.15 (although I see that 0.7.18 is now available)
The NeDB database export completed successfully.
I then changed my config.json to utilize Mongo. 

When importing the exported database in to MongoDB these two errors were thrown (not sure if these errors have anything to do with the trouble I'm now seeing though).

```
C:\Program Files\Open Source\MeshCentral>node node_modules/meshcentral --dbimport myserverdb.json
ERR: the options [useNewUrlParser] is not supported
ERR: the options [useUnifiedTopology] is not supported
Imported 2116 objects(s) from myserverdb.json.
```

I restarted MeshCentral after importing the database and can tell it pulled in my data as my logon was now working and all my groups and devices showed up.
However, I can now no longer connect to any of the devices in any fashion.  The screen grab below is what I am seeing- all the connection options (like Desktop, Terminal, etc) are no longer present.
I thought maybe I needed to wait a bit for the data to populate from the agents but that doesn't seem to be the case.  
![mesh](https://user-images.githubusercontent.com/48607498/101716405-30d35500-3a6b-11eb-865b-ce959032afb9.JPG)

Any ideas what I should be looking at next?  
Also, where is the meshcentral MongoDB data stored?  I installed MongoDB using the default options.  Lastly, does the MeshCentral backup process also include the MongoDB database as part of the backup job?

Edit: I'm also seeing those 'useNewUrlParser and useUnifiedTopology is not supported' errors in mesherrors.txt when I start the server using MongoDB.  So maybe those are related to the issue I'm seeing.

**Comment by PathfinderNetworks** [comment]
BTW, I rolled back to using nedb for now (by just changing config.json back to the default nedb config) and everything is working.  So I'm not down or anything- but would like to get MongoDB figured out so I can use that instead.  I have over 300 devices connected to MeshCentral so I'm pushing the limits with NeDB.

**Comment by PathfinderNetworks** [comment]
One other thing, I didn't realize I had to manually create the C:\data\db folder for Mongo.  So that folder didn't exist when I imported the meshcentral database in to Mongo.  I have since created that folder and restarted Mongo and Mesh and saw that Mongo populated the db folder with a number of files.
However, it's still not working correctly- still have the same issue I mention above.  

BTW, the install guide at https://info.meshcentral.com/downloads/MeshCentral2/MeshCentral2InstallGuide-0.1.0.pdf doesn't discuss how to install, setup, or use MongoDB in the Windows install section.  That would be hugely beneficial if that guide included the steps needed to use Mongo.

**Comment by Ylianst** [comment]
Hi. Can you let me know what version of MongoDB you installed and send over your exact MongoDB configuration you tried in config.json? (Do put XXXXX on any private data). Thanks.

**Comment by PathfinderNetworks** [comment]
It is the current Windows version of MongoDB - 4.4.2
Here is a snippet of the setting section of my config.json

   "settings": {
	  "MongoDb": "mongodb://127.0.0.1:27017/meshcentral",
      "RedirPort": 80, 
      "cert": "mesh.xxxxxxxxx.com", 
      "wanonly": true, 
      "selfupdate": false,
	  "desktopmultiplexor": true,
	  "WsCompression": true,
	  "AgentWsCompression": true,
	  "autoBackup": {

**Comment by Ylianst** [comment]
Ok. That seems pretty standard. Let me check and get back.

**Comment by PathfinderNetworks** [comment]
Hi Ylian, just wondering if you had a chance to look at this?  No rush, but is something I'd like to get figured out.

**Comment by PathfinderNetworks** [comment]
I tried moving to MongoDB again this evening. Still getting the same issue as mentioned above.  But I do have a bit more info.  When I go to view my Device Groups I notice that my logon (which is the only logon on this MeshCentral server) has 'No Rights' to any of the groups.  I'm guessing whatever is the reason behind that is why I am unable to connect to any of the devices.  It's as though the database export/import isn't grabbing all the data from the NeDB database and moving it over in to the MongDB?

**Comment by PathfinderNetworks** [comment]
One other thing- when I view one of the existing groups I can see that there are no users authorized.  Nor is the 'Add Users' link present.  But, if I look at the test group I created, it does show my logon in the 'User Authorizations' are and the Add Users link is present as well.

**Comment by PathfinderNetworks** [comment]
Just wanted to keep this issue active.  
Any ideas about why the user permissions may not be moving over correctly when migrating from NeDB to MongoDB?  I think that's the root issue I'm facing in trying to get away from NeDB.

**Comment by Ylianst** [comment]
Hi. So sorry for the delay. For the following errors:

```
ERR: the options [useNewUrlParser] is not supported
ERR: the options [useUnifiedTopology] is not supported
```

This is probably because your server has a very old version of the NodeJS module that interacts with the MongoDB database. Can you stop your server, rename the "node_modules" folder to "node_modules_bak" and reinstall MeshCentral like this:

```
mv node_modules node_modules_bak
npm install meshcentral
node node_modules/meshcentral
```

This will run MeshCentral again with updated modules. Even if you keep NeDB, I would do this just to make sure you have the latest dependencies.

I am going to run a test import/export now and see if I can find any issues.
This other thing is that I am not completely sure

**Comment by PathfinderNetworks** [comment]
Thanks Ylian,
I performed the steps you mentioned.  But now the MeshCentral service will not start.
I am getting this error:

internal/modules/cjs/loader.js:1083
  throw err;
  ^

Error: Cannot find module 'c:\Program Files\Open Source\MeshCentral\node_modules\node-windows\lib\wrapper.js'
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:1080:15)
    at Function.Module._load (internal/modules/cjs/loader.js:923:27)
    at Function.executeUserEntryPoint [as runMain] (internal/modules/run_main.js:72:12)
    at internal/main/run_main_module.js:17:47 {
  code: 'MODULE_NOT_FOUND',
  requireStack: []
}

Looking at the new node_modules folder I can see there is no Lib directory there at all.  
Is there some different step/process to set up MeshCentral as a Windows service?

**Comment by PathfinderNetworks** [comment]
Digging a bit deeper I can see that I can't even start MeshCentral at all- let alone as a service. When trying to manually run it I am getting this error:

Windows Script Host
Script:  C:\Program Files\Open Source\MeshCentral\node_modules\meshcentral\meshcentral.js
Line: 114
Char: 38
Error: Syntax Error
Code: 800A03EA
Source: Microsoft JScript compilation error

**Comment by PathfinderNetworks** [comment]
I'm closing this issue as the original trouble reported has been resolved under issue #2128 
I opened a new issue with just the discussion about the node_modules trouble under #2131
